addons/hr_expense/models/hr_expense.py

```diff
@@ -318,6 +318,10 @@ def _compute_analytic_distribution(self):
             })
             expense.analytic_distribution = distribution or expense.analytic_distribution
 
+    def _inverse_total_amount(self):
+        for expense in self:
+            expense.unit_amount = expense.total_amount_company
```

Comment:
```suggestion
            expense.unit_amount = expense.total_amount_company / expense.quantity
```
As discussed, debated and thought deeply (including the PO)
 - The fix as it is fine (would  make the `unit_amount = total_amout / quantity` in the inverse) even though it shouldn't change much anyway
 - The refactor of the module is getting increasingly more urgent, thank you for getting me more arguments in that fight :p

Reply:
Thanks @JulienAlardot!
I have committed the suggested changes.